### PR TITLE
Flav/add assistant quick edition menu

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

See https://github.com/dust-tt/dust/issues/3262.

This PR adds a new component `AssistantQuickEditionMenu` that will be displayed in the conversation view. I took the opportunity to factorize some duplicated code to add an assistant to the user list as well.

AssistantQuickEditionMenu

## Risk

All those changes are still gated to the Dust team only.

### CODE REVIEW

File: front/components/assistant/conversation/AgentMessage.tsx [hunk 1]
```diff
@@ -316,27 +317,33 @@ export function AgentMessage({
...
       enableEmojis={true}
       renderName={() => {
         return isDevelopmentOrDustWorkspace(owner) ? (
-          <div className="text-sm font-medium">
-            {AssitantDetailViewLink(agentMessageToRender.configuration)}
+          <div className="flex flex-row gap-2">
+            <div className="text-sm font-medium">
+              {AssitantDetailViewLink(agentConfiguration)}
+            </div>
+            <AssistantQuickEditionMenu
```
Comment (philipperolet) [inline front/components/assistant/conversation/AgentMessage.tsx:340]: nit: AssistantEditionMenu 
Also, :pray: will reuse for 3265 rn

File: front/lib/client/dust_api.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import type { AgentUserListStatus, WorkspaceType } from "@dust-tt/types";
+
+import type { PostAgentListStatusRequestBody } from "@app/pages/api/w/[wId]/members/me/agent_list_status";
+
+interface PerformAgentUserListStatusUpdateParams {
```
Comment (philipperolet) [inline front/lib/client/dust_api.ts:5]: nit: cf first com, would rename UpdateAgentUserListStatus (a tad less verbose)

File: front/lib/client/dust_api.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  owner: WorkspaceType;
+  agentConfigurationSId: string;
+}
+
+interface PerformAgentUserListStatusUpdateResult {
+  success: boolean;
+  errorMessage?: string;
+}
+
+export async function performAgentUserListStatusUpdate({
```
Comment (philipperolet) [inline front/lib/client/dust_api.ts:16]: and here too

File: types/src/front/assistant/agent.ts [hunk 4]
```diff
@@ -152,3 +152,21 @@ export type AgentConfigurationType = LightAgentConfigurationType & {
   // performed (potentially NoOp eg autoSkip above).
   action: AgentActionConfigurationType | null;
 };
+
+export function isGlobalAgentConfiguration(
```
Comment (philipperolet) [inline types/src/front/assistant/agent.ts:156]: IMHO we don't need those 3 functions, the scope test is clear enough and short enough; available to discuss if needed

Wrapping lower level logic in functions is fine if either it dries enough, or it makes clearer enough, here IMHO not the case

Comment (philipperolet) [reply types/src/front/assistant/agent.ts:156]: the varying names are confusing (isglobal, then isagentinworkspace, then isagentPublished

File: front/components/assistant/conversation/AssistantQuickEditionMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+interface AssistantQuickEditionMenuProps {
+  agentConfigurationSId: string;
+  owner: WorkspaceType;
+}
+
+export function AssistantQuickEditionMenu({
+  // The `agentConfiguration` cannot be used directly as it isn't dynamically
+  // updated upon user list mutations. This limitation stems from its
+  // propagation method from <ConversationMessage>.
+  agentConfigurationSId,
```
Comment (philipperolet) [inline front/components/assistant/conversation/AssistantQuickEditionMenu.tsx:32]: To be confirmed, but I think spolu told me to call that agentConfigId (not sId) => in the upper front layer, there are no model ids, all ids should be sIds (cf just below here 'workspaceId')

File: front/components/assistant/conversation/AssistantQuickEditionMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+  if (isGlobalAgentConfiguration(agentConfiguration)) {
+    return <></>;
+  }
+
+  const isAgentInWorkspace =
+    isAgentConfigurationInWorkspace(agentConfiguration);
+  const isAgentPublished = isAgentConfigurationPublished(agentConfiguration);
+
+  const isInList = agentConfiguration.userListStatus === "in-list";
```
Comment (philipperolet) [inline front/components/assistant/conversation/AssistantQuickEditionMenu.tsx:56]: nit: cf below, I'm usually in favor of doing that, but here it's adding code for something that's already clear enough

very nit though :)

File: front/components/assistant/conversation/AssistantQuickEditionMenu.tsx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    return <></>;
+  }
+
+  if (isGlobalAgentConfiguration(agentConfiguration)) {
+    return <></>;
+  }
+
+  const isAgentInWorkspace =
+    isAgentConfigurationInWorkspace(agentConfiguration);
+  const isAgentPublished = isAgentConfigurationPublished(agentConfiguration);
```
Comment (philipperolet) [inline front/components/assistant/conversation/AssistantQuickEditionMenu.tsx:54]: cf nit below

**Review by philipperolet — CHANGES_REQUESTED** [review]
Looks great ! Would remove the 3 functions last (let's chat if you disagree), otherwise good with minor nits. Thanks for the menu code :pray:

File: front/components/assistant/conversation/AssistantQuickEditionMenu.tsx (hunk 7 above)
Comment (philipperolet) [reply front/components/assistant/conversation/AssistantQuickEditionMenu.tsx:54]: (or nit above :sweat_smile: )

File: front/components/assistant/conversation/AgentMessage.tsx (hunk 1 above)
Comment (flvndvd) [reply front/components/assistant/conversation/AgentMessage.tsx:340]: okok

File: types/src/front/assistant/agent.ts (hunk 4 above)
Comment (flvndvd) [reply types/src/front/assistant/agent.ts:156]: I went with this approach following our discussion from yesterday, were I feel like we the scope is slightly changing with the ongoing refactoring. But I agree with the "dry" point and I'm happy to remove it.

Comment (flvndvd) [reply types/src/front/assistant/agent.ts:156]: I'm happy to remove but will probably push for it if one day is published becomes something like:
```
["published", "workspace].include(agentConfiguration.scope);
```

🙃

File: front/components/assistant/conversation/AssistantQuickEditionMenu.tsx (hunk 5 above)
Comment (flvndvd) [reply front/components/assistant/conversation/AssistantQuickEditionMenu.tsx]: Ok good to know, let me update.

File: types/src/front/assistant/agent.ts (hunk 4 above)
Comment (philipperolet) [reply types/src/front/assistant/agent.ts]: absolutely agree. As soon as the logic become more complex than just " scope === X" then the functions make sense

Comment (philipperolet) [reply types/src/front/assistant/agent.ts]: but IMHO we should wait until that happens

**Review by philipperolet — APPROVED** [review]
:+1:

**Pull request merged**